### PR TITLE
update optimistic transaction

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by kissmydb** [comment]
@zhenjiaogao

File: optimistic-transaction.md [hunk 1]
```diff
@@ -114,7 +112,7 @@ tidb_retry_limit = 10
 
 > **注意：**
 >
-> `tidb_retry_limit` 变量决定了事务重试的最大次数。当它被设置为 0 时，所有事务都不会自动重试，包括自动提交的单语句隐式事务。这是彻底禁用 TiDB 中自动重试机制的方法。禁用自动重试后，所有冲突的事务都会以最快的方式上报失败信息 (`try again later`) 给应用层。
+> `tidb_retry_limit` 变量决定了事务重试的最大次数。当它被设置为 0 时，所有事务都不会自动重试，包括自动提交的单语句隐式事务。这是彻底禁用 TiDB 中自动重试机制的方法。禁用自动重试后，所有冲突的事务都会以最快的方式上报失败信息 (包含 `try again later`) 给应用层。
```
Comment (ran-huang) [inline optimistic-transaction.md:115]:
```suggestion
> `tidb_retry_limit` 变量决定了事务重试的最大次数。当它被设置为 0 时，所有事务都不会自动重试，包括自动提交的单语句隐式事务。这是彻底禁用 TiDB 中自动重试机制的方法。禁用自动重试后，所有冲突的事务都会以最快的方式上报失败信息（包含 `try again later`）给应用层。
```

File: optimistic-transaction.md [hunk 2]
```diff
@@ -132,34 +130,9 @@ TiDB 默认不进行事务重试，因为重试事务可能会导致更新丢失
...
-capacity = 2048000
-```
-
-配置项 `capacity` 主要影响到冲突判断的正确性。在实现冲突检测时，不可能把所有的 Key 都存到内存里，所以真正存下来的是每个 Key 的 Hash 值。有 Hash 算法就有碰撞也就是误判的概率，这里可以通过配置 `capacity` 来控制 Hash 取模的值：
-
-* `capacity` 值越小，占用内存小，误判概率越大。
-* `capacity` 值越大，占用内存大，误判概率越小。
-
-实际应用时，如果业务场景能够预判断写入不存在冲突（如导入数据操作），建议关闭冲突检测。
+作为一个分布式系统，TiDB 在内存中的冲突检测是在 TiKV 中进行。主要发生在 prewrite 阶段。因为 TiDB 集群是一个分布式系统，TiDB 实例本身无状态，实例之间无法感知到彼此的存在，也就无法确认自己的写入与别的 TiDB 实例是否存在冲突，所以会在 TiKV 这一层检测具体的数据是否有冲突。
```
Comment (ran-huang) [inline optimistic-transaction.md:133]:
```suggestion
作为一个分布式系统，TiDB 在内存中的冲突检测是在 TiKV 中进行，主要发生在 prewrite 阶段。因为 TiDB 集群是一个分布式系统，TiDB 实例本身无状态，实例之间无法感知到彼此的存在，也就无法确认自己的写入与别的 TiDB 实例是否存在冲突，所以会在 TiKV 这一层检测具体的数据是否有冲突。
```

File: optimistic-transaction.md [hunk 3]
```diff
@@ -7,36 +7,36 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-optimistic/','/docs-c
...
-本文介绍 TiDB 乐观事务的原理，以及相关特性。本文假定你对 [TiDB 的整体架构](/architecture.md#tidb-整体架构)、[Percolator](https://www.usenix.org/legacy/event/osdi10/tech/full_papers/Peng.pdf) 事务模型以及事务的 [ACID 特性](/glossary.md#acid)都有一定了解。
+本文介绍 TiDB 乐观事务的原理，以及相关特性。
 
-TiDB 默认使用乐观事务模型，不会出现读写冲突，所有的读操作都不会被写操作阻塞。对于写写冲突，只有在客户端执行 `COMMIT` 时，才会触发两阶段提交并检测是否存在写写冲突。
+TiDB 默认使用乐观事务模型。对于写写冲突，只有事务提交时，才会通过两阶段提交检测是否存在写写冲突。
 
 > **注意：**
 >
-> 自 v3.0.8 开始，TiDB 默认使用[悲观事务模型](/pessimistic-transaction.md)。但如果从 3.0.7 及之前的版本升级到 >= 3.0.8 的版本，不会改变默认事务模型，即**只有新创建的集群才会默认使用悲观事务模型**。
+> 自 v3.0.8 开始，新创建的 TiDB 集群默认使用[悲观事务模型](/pessimistic-transaction.md)。但如果从 3.0.7 版本及之前的创建的集群升级到 >= 3.0.8 的版本，不会改变默认事务模型，即**只有新创建的集群才会默认使用悲观事务模型**。
```
[the remark this answers is not in the document]
Comment (zhenjiaogao) [reply optimistic-transaction.md:16]:
```suggestion
> 自 v3.0.8 开始，新创建的 TiDB 集群默认使用[悲观事务模型](/pessimistic-transaction.md)。但如果从 v3.0.7 版本及之前创建的集群升级到 >= v3.0.8 的版本，不会改变默认的事务模型，即**只有新创建的集群才会默认使用悲观事务模型**。
```

File: optimistic-transaction.md [hunk 4]
```diff
@@ -7,36 +7,36 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-optimistic/','/docs-c
...
 
 3. 客户端发起写请求。
 
-    TiDB 校验写入数据是否符合一致性约束（如数据类型是否正确、是否符合唯一索引约束等）。**校验通过的数据将存放在内存里。**
+    TiDB 校验写入数据是否符合约束（如数据类型是否正确、是否符合非空约束等）。**校验通过的数据将存放在 TiDB 中该事务的私有内存里。**
 
 4. 客户端发起 commit。
 
-5. TiDB 开始两阶段提交，保证分布式事务的原子性，让数据真正落盘。
+5. TiDB 开始两阶段提交，保证事务的原子性的前提下，数据持久化。
```
Comment (zhenjiaogao) [inline optimistic-transaction.md:39]:
```suggestion
5. TiDB 开始两阶段提交，在保证事务原子性的前提下，进行数据持久化。
```

**Comment by ran-huang** [comment]
@coocood PTAL. Thanks~

**Comment by jackysp** [comment]
PTAL @zhenjiaogao

File: optimistic-transaction.md [hunk 5]
```diff
@@ -61,18 +61,16 @@ TiDB 中事务使用两阶段提交，流程如下：
...
-
 ## 事务的重试
 
-使用乐观事务模型时，在高冲突率的场景中，事务很容易提交失败。而 MySQL 内部使用的是悲观事务模型，在执行 SQL 语句的过程中进行冲突检测，所以提交时很难出现异常。为了兼容 MySQL 的悲观事务行为，TiDB 提供了重试机制。
+使用乐观事务模型时，在高冲突率的场景中，事务容易发生写写冲突而导致提交失败。MySQL 使用悲观事务模型，在执行 SQL 语句的过程中进行加锁并且其 Repeatable Read 隔离级别允许出现不可重复读，所以提交时一般不会出现异常。为了降低应用改造难度，TiDB 提供了数据库内部自动重试机制。
 
 ### 重试机制
 
-当事务提交后，如果发现冲突，TiDB 内部重新执行包含写操作的 SQL 语句。你可以通过设置 `tidb_disable_txn_auto_retry = off` 开启自动重试，并通过 `tidb_retry_limit` 设置重试次数：
+当事务提交时，如果发现写写冲突，TiDB 内部重新执行包含写操作的 SQL 语句。你可以通过设置 `tidb_disable_txn_auto_retry = off` 开启自动重试，并通过 `tidb_retry_limit` 设置重试次数：
```
Comment (cfzjywxk) [inline optimistic-transaction.md:73]: 这里是否需要提一下开启重试会破坏隔离级别

Comment (jackysp) [reply optimistic-transaction.md:73]: https://github.com/pingcap/docs-cn/blob/85afc6de27c1d7b4e171558a06ffe55489742e98/optimistic-transaction.md#%E9%87%8D%E8%AF%95%E7%9A%84%E5%B1%80%E9%99%90%E6%80%A7
下面重试的局限性有介绍

**Comment by coocood** [comment]
LGTM

**Pull request merged**